S7.Net/PLCExceptions.cs

```diff
@@ -69,7 +70,7 @@ protected InvalidVariableTypeException(SerializationInfo info, StreamingContext
         #endif
     }
 
-    internal class TPKTInvalidException : Exception
+    internal class TPKTInvalidException : IOException
```

Comment:
I don't agree on this. An `IOException` in my opionion is an exception because in or out failed, in this case it didn't fail, it just returned something unexpected. It also doesn't matter much what the underlying exception was, when we can't read the TPKT or TPDU the connection is messed up and we need to close it. It would also help if the connection send/receive was used in a single place so we could standardize there.

Reply:
I agree with your argumentation, and will revert this to inherit from System.Exception.

In effect this means that I also need to check for those 2 exception in the places where I close the connection - since as you said the S7 connection is somehow messed up. As you said it would be great if there were fewer places where this would have to happen, but I don't think one can hope for the code to clean itself up all by itself, so in the meantime there would be a bit of duplication.

I guess the most likely use case for all of this is when people use a plc connection in a unsynchronized way from multiple threads. So maybe all of this is not worth it at all :)